string robot_controller
---
string robot_result